controllers/admin/AdminCartRulesController.php

```diff
@@ -681,6 +715,24 @@ public function renderForm()
         $cart_rules = $current_object->getAssociatedRestrictions('cart_rule', false, true, 0, $limit);
         $carriers = $current_object->getAssociatedRestrictions('carrier', true, true);
 
+        // If the cart rule applies to all carts then display a warning
+        if (empty($current_object->code) &&
+            empty($current_object->country_restriction) &&
+            empty($current_object->carrier_restriction) &&
+            empty($current_object->group_restriction) &&
+            empty($current_object->cart_rule_restriction) &&
+            empty($current_object->product_restriction) &&
+            empty($current_object->id_customer) &&
+            $current_object->minimum_amount == 0 &&
+            $current_object->date_to > date('Y-m-d H:i:s')
```

Comment:
Same comments about `date`.

Ps. Is it normal you use `date_to` instead of `date_from` here ?

Reply:
> Is it normal you use date_to instead of date_from here ?

Yes, because we will display the message only if the voucher is not yet expired.